weld/subsystem/src/main/java/org/jboss/as/weld/deployment/processors/BuildCompatibleExtensionProcessor.java

```diff
@@ -0,0 +1,126 @@
+            method = clazz.getDeclaredMethod("getBuildCompatibleExtensions");
+            log.debugf("Found getBuildCompatibleExtensions for %s", BUILD_COMPATIBLE_EXTENSION_LOADER);
+        } catch (ClassNotFoundException | NoSuchMethodException e) {
+            // ignore
+            log.debugf(e, "Cannot find getBuildCompatibleExtensions method in %s", BUILD_COMPATIBLE_EXTENSION_LOADER);
+        }
+        getBuildCompatibleExtensions = method;
+    }
+
+    /**
+     * Reflectively calls the BuildCompatibleExtensionLoader#getBuildCompatibleExtensions to see
+     * if BuildCompatibleExtensions existing in the current deployment. This requies that the
+     * thread context class loader is set to the deployment unit module ClassLoader.
+     * @return true if getBuildCompatibleExtensions returns a list > 0, false otherwise
+     * @throws DeploymentUnitProcessingException - on reflection invocation failure
+     */
+    static boolean hasBuildCompatibleExtensions() throws DeploymentUnitProcessingException {
+        if (getBuildCompatibleExtensions != null) {
+            try {
+                List extensions = (List) getBuildCompatibleExtensions.invoke(null);
+                return extensions.size() > 0;
+            } catch (ReflectiveOperationException e) {
+                throw new DeploymentUnitProcessingException(e);
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public void deploy(DeploymentPhaseContext phaseContext) throws DeploymentUnitProcessingException {
+        final DeploymentUnit deploymentUnit = phaseContext.getDeploymentUnit();
+        if (deploymentUnit.getParent() != null) {
+            return; // only run for top-level deployments
+        }
+
+        final Module module = deploymentUnit.getAttachment(org.jboss.as.server.deployment.Attachments.MODULE);
+        final ClassLoader classLoader = module.getClassLoader();
+        final ClassLoader currentClassLoader = Thread.currentThread().getContextClassLoader();
+        try {
+            // Have to set the deployment module class loader as the build compatible extension
+            // loader and translator use the TCCL to search for BuildCompatibleExtensions using
+            // SerivceLoader
+            setContextClassLoader(classLoader);
+            if(hasBuildCompatibleExtensions()) {
+                // there are build compatible extensions visible, register the translator
+                WeldPortableExtensions.getPortableExtensions(deploymentUnit)
+                        .tryRegisterExtension(loadClass(LITE_EXTENSION_TRANSLATOR, classLoader),
+                                              deploymentUnit);
+                log.debugf("Registered %s for %s", LITE_EXTENSION_TRANSLATOR, deploymentUnit.getName());
+            }
+        } finally {
+            setContextClassLoader(currentClassLoader);
+        }
+    }
+
+    private void setContextClassLoader(ClassLoader classLoader) {
```

Comment:
WildFlySecurityManager.setCurrentContextClassLoaderPrivileged can handle the TCCL stuff.

Reply:
I have switched to use that.